onadata/libs/utils/logger_tools.py

```diff
@@ -797,6 +798,19 @@ class OpenRosaResponseForbidden(OpenRosaResponse):
     status_code = 403
 
 
+class OpenRosaNotAuthenticated(Response):
+    status_code = 401
+
+    def __init__(self, *args, **kwargs):
+        super(OpenRosaNotAuthenticated, self).__init__(*args, **kwargs)
+
+        self['Content-Type'] = 'text/html; charset=utf-8'
+        self['X-OpenRosa-Accept-Content-Length'] = DEFAULT_CONTENT_LENGTH
+        tz = pytz.timezone(settings.TIME_ZONE)
+        dt = datetime.now(tz).strftime('%a, %d %b %Y %H:%M:%S %Z')
+        self['Date'] = dt
+
+
```

Comment:
What challenges were there when subclassing `BaseOpenRosaResponse` instead which already as 807 to 811 implemented?

Reply:
The `BaseOpenRosaResponse`  which handles all the other OpenRosa exceptions inherits from django's HttpResponse class.
The response object for requests failing with 401 status code should be an instance of django's default response class.

Another striking difference is the content_type.
The other exceptions would have the response content type set to `text/xml; charset=utf-8` the NotAuthenticated OpenRosa exception requires the content type to be`text/html; charset=utf-8`

Reply:
@WinnyTroy Why does it need to be `text/html`? If it needs to be HTML the name might be a bit confusing since the OpenRosa standard mostly responds in `XML`.

Also, why does it need to be a Django response in particular ?

Reply:
Before implementing this class to include the headers, the response object being returned was an instance of django's default response class. The content type for this response object was `text/html`. Keeping in line with existing implementation flow, I didnt want to change that to xml

Also, trying to implement this class using django's HttpResponse class breaks existing tests with `AttributeError: 'OpenRosaNotAuhenticated' object has no attribute 'request'`. So that is why i subclassed this from django's default response class.